overlord/hookstate/hookmgr.go

```diff
@@ -143,13 +153,38 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	// TODO: Actually dispatch the hook.
+	_, err = RunHook(setup.Snap, setup.Revision, setup.Hook)
+	if err != nil {
+		if handlerErr := handler.Error(err); handlerErr != nil {
+			return handlerErr
+		}
+
+		return err
+	}
 
-	// Done with the hook. TODO: Check the result, if success call Done(), if
-	// error, call Error(). Since we have no hooks, for now we just call Done().
-	if err := handler.Done(); err != nil {
+	if err = handler.Done(); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func doRunHook(snapName string, revision snap.Revision, hookName string) ([]byte, error) {
+	revisionString := ""
+	if !revision.Unset() {
+		revisionString = revision.String()
+	}
+
+	return runCommand("snap", "run", snapName, "--hook", hookName, "-r", revisionString)
+}
+
+func doRunCommand(name string, args ...string) ([]byte, error) {
+	output, err := exec.Command(name, args...).CombinedOutput()
+	if err != nil {
+		// "exit code 1" is not particularly helpful. So rewrite the error
+		// using stdout and stderr instead.
+		err = fmt.Errorf("%s", strings.Trim(string(output), " \n"))
```

Comment:
I think we want to give some more context to the error here. If for example a command just exits with exist-status 7 but no output the error would become "". Maybe more like: `"fmt.Errorrf("failed to run command %q %q: %q (%s)", name, args, output, err)` or similar, I don't mind the exact message, but I would like to see the original error (permission denied, command not found, exit status N etc) plus what command it actually was that run and the output.